### PR TITLE
ci(orc8r): Added Terraform Lint CI Check

### LINKED ISSUE #11996: Create CI job for checking terraform format checks
Setup CI on GitHub Actions for checking terraform format on all `.tf` files with the following command. Also check if we have something in marketplace that we can use. 
```bash
terraform fmt
```

### PR DESCRIPTION
**Pull request by ShubhamTatvamasi**
Signed-off-by: Shubham Tatvamasi <shubhamtatvamasi@gmail.com>

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Review by themarwhal — APPROVED** [review]
lgtm!

File: .github/workflows/terraform-lint.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+on:
+  push:
+    branches:
+      - master
+      - 'v1.*'
+  pull_request:
+    branches:
+      - master
+      - 'v1.*'
+
```
Comment (themarwhal) [inline .github/workflows/terraform-lint.yml:13]: should we add some concurrency configs? @quentinDERORY

**Comment by ShubhamTatvamasi** [comment]
For common tasks such as lint check, should we start using actions from marketplace?
https://github.com/marketplace/actions/terraform-lint-action

**Comment by ShubhamTatvamasi** [comment]
I have updated code, now using action from marketplace.

**Review by themarwhal — APPROVED** [review]

**Review by Neudrino — COMMENTED** [review]
> For common tasks such as lint check, should we start using actions from marketplace? https://github.com/marketplace/actions/terraform-lint-action

I agree that using Actions from Marketplace might be a good idea.
However, on top of that we might want to stick to the standard [ReviewDog](https://github.com/reviewdog/reviewdog/blob/master/README.md) provided linters where possible, as they provide excellent integration with Github out of the box, and avoid custom code here.

In particular for this case this would mean to us [tflint](https://github.com/marketplace/actions/run-tflint-with-reviewdog).
This would also have the advantage that it could also easily be integrated with [our existing set (and setup) of reviewdog](https://github.com/magma/magma/blob/master/.github/workflows/reviewdog-workflow.yml).

File: .github/workflows/terraform-lint.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+---
+name: Terraform Lint
+
+on:
```
Comment (Neudrino) [inline .github/workflows/terraform-lint.yml]: yamllint...
```suggestion
on:  # yamllint disable-line rule:truthy
```

**Review by Neudrino — CHANGES_REQUESTED** [review]

**Review by Neudrino — APPROVED** [review]
More linters are always good and this PR LGTM in general.

We might consider to us [tflint](https://github.com/marketplace/actions/run-tflint-with-reviewdog) and integrate it with our [our existing set (and setup) of reviewdog](https://github.com/magma/magma/blob/master/.github/workflows/reviewdog-workflow.yml) linters to benefit from the excellent Github integration of reviewdog :dog:.

File: .github/workflows/terraform-lint.yml (hunk 1 above)
Comment (Neudrino) [reply .github/workflows/terraform-lint.yml:13]:
```yaml
concurrency:
  group: ${{ github.workflow }}-${{ github.event.pull_request.number}}
  cancel-in-progress: true
```
Maybe something like this in alignment with the other linters.

**Pull request merged**